### PR TITLE
feat: changed proptype from `string`/`IconSubComponentProps` to `SubIcon`

### PR DESCRIPTION
**Pull request by MohammedHaris96**
#### Basic

- [ ] Used plop (`npm run plop`) to create a new component.
- [ ] PR has description.
- [ ] New component is functional and uses Hooks.
- [ ] Component is written in TypeScript.

#### Style

- [ ] Styles are added to `NewComponent.modules.scss` file inside the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/).
- [ ] Styles are defined using [monday-ui-style](https://github.com/mondaycom/monday-ui-react-core/tree/monorepo-style/packages/monday-ui-style) tokens
- [ ] Displays correctly in all the themes

#### Storybook

- [ ] Stories were added to `/src/components/NewComponent/__stories__/NewComponent.mdx` file.
- [ ] Stories include all flows of using the component.
- [ ] Stories implemented using [vibe-storybook-components](https://github.com/mondaycom/vibe-storybook-components) components and tokens.

#### Tests

- [ ] Tests are compliant with [TESTING_README.md](/packages/core/TESTING_README.md) instructions.

#### Accessibility

- [ ] Component is accessible.
- [ ] Component is compatible with [Vibe Accessibility Guidelines](https://style.monday.com/?path=/docs/foundations-accessibility--docs)

### CODE REVIEW

File: packages/core/src/components/Modal/ModalHeader/ModalHeader.tsx [hunk 1]
```diff
@@ -12,6 +12,7 @@ import Text from "../../Text/Text";
 import Heading from "../../Heading/Heading";
 import Flex from "../../Flex/Flex";
 import styles from "./ModalHeader.module.scss";
+import { SubIcon } from "src/types";
```
Comment (YossiSaadi) [inline packages/core/src/components/Modal/ModalHeader/ModalHeader.tsx:15]: I'm not sure how it behaves
Can you, for the meantime, change it to a relative path? same as other imports in the file? (`../../../types`)

File: packages/core/src/components/Tabs/Tab/Tab.tsx [hunk 2]
```diff
@@ -4,11 +4,12 @@ import React, { FC, forwardRef, ReactElement, useRef } from "react";
 import { noop as NOOP } from "lodash-es";
 import useMergeRef from "../../../hooks/useMergeRef";
 import { getStyle } from "../../../helpers/typesciptCssModulesHelper";
-import Icon, { IconSubComponentProps } from "../../Icon/Icon";
+import Icon from "../../Icon/Icon";
 import VibeComponentProps from "../../../types/VibeComponentProps";
 import { IconType } from "../../Icon/IconConstants";
 import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";
 import styles from "./Tab.module.scss";
+import { SubIcon } from "src/types";
```
Comment (YossiSaadi) [inline packages/core/src/components/Tabs/Tab/Tab.tsx:12]: Same
Can you change it to a relative path? same as other imports in the file? (`../../../types`)

**Review by YossiSaadi — COMMENTED** [review]
Great! can you please change to a relative path?
I'm not sure it would build fine with the current impl

File: packages/core/src/components/Modal/ModalHeader/ModalHeader.tsx (hunk 1 above)
Comment (MohammedHaris96) [reply packages/core/src/components/Modal/ModalHeader/ModalHeader.tsx:15]: Sure @YossiSaadi, allow me some time I'll raise another PR.

File: packages/core/src/components/Tabs/Tab/Tab.tsx (hunk 2 above)
Comment (MohammedHaris96) [reply packages/core/src/components/Tabs/Tab/Tab.tsx:12]: Sure @YossiSaadi, allow me some time I'll raise another PR.

**Comment by YossiSaadi** [comment]
no need for another PR, you can commit another commit to this PR :)

**Comment by YossiSaadi** [comment]
resolves #2025

**Comment by YossiSaadi** [comment]
hope you don't mind, I've made a little change to the title :)
- we want it to be declared as a `feat` and not necessarily a `fix`, since it introduces an ability and not fixing a bug for the end user
- change the type change to be a bit more specific

Let's see how the CI goes and then we'll be able to merge.
Great job on your first contribution to Vibe!

**Comment by MohammedHaris96** [comment]
Thank you so much @YossiSaadi  for 

> hope you don't mind, I've made a little change to the title :)
> 
> * we want it to be declared as a `feat` and not necessarily a `fix`, since it introduces an ability and not fixing a bug for the end user
> * change the type change to be a bit more specific
> 
> Let's see how the CI goes and then we'll be able to merge. Great job on your first contribution to Vibe!

Thank you so much @YossiSaadi for addressing this. :)

**Pull request merged**